data/modules/BreakdownServicing/BreakdownServicing.lua

```diff
@@ -284,16 +295,14 @@ local onEnterSystem = function (ship)
 			service_history.jumpcount = service_history.jumpcount - fixup
 		else
 			-- Destroy the engine
-			local engine = ship:GetEquip('engine',1)
-
 			if engine.fuel.name == 'military_fuel' then
 				pigui.playSfx("Hyperdrive_Breakdown_Military", 1.0, 1.0)
 			else
 				pigui.playSfx("Hyperdrive_Breakdown", 1.0, 1.0)
 			end
 
-			ship:RemoveEquip(engine)
-			ship:GetComponent('CargoManager'):AddCommodity(Commodities.rubbish, engine.capabilities.mass)
+			ship:GetComponent('EquipSet'):Remove(engine)
+			ship:GetComponent('CargoManager'):AddCommodity(Commodities.rubbish, engine.mass)
 
 			Comms.Message(l.THE_SHIPS_HYPERDRIVE_HAS_BEEN_DESTROYED_BY_A_MALFUNCTION)
```

Comment:
Discussion;

Wow, this seems pretty excessive.. no chance of a (repairable) malfunction?  (I appreciate this is merely duplicating existing behaviour..)

Reply:
It's in the modules folder; modules can do what they want. We currently don't have the framework to express a repairable-after-the-fact malfunction, and none of the supporting systems to handle an "installed but not functional" equipment item.

The plan (loosely held) is for this to be replaced with some code that sets the drive's integrity to 0, rendering it unusable until repaired or in-situ replaced.